drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -0,0 +1,927 @@
+{
+	usb_device_callback_message_struct_t *mcux_msg = msg;
+	uint8_t ep;
+	usb_device_notification_t mcux_notify;
+	struct udc_mcux_data *priv;
+	const struct device *dev;
+	usb_status_t mcux_status = kStatus_USB_Success;
+	int err = 0;
+
+	if ((NULL == msg) || (NULL == handle)) {
+		return kStatus_USB_InvalidHandle;
+	}
+
+	mcux_notify = (usb_device_notification_t)mcux_msg->code;
+	priv = (struct udc_mcux_data *)(PRV_DATA_HANDLE(handle));
+	dev = priv->dev;
+
+	switch (mcux_notify) {
+	case kUSB_DeviceNotifyBusReset:
+		struct udc_ep_config *cfg;
+
+		udc_mcux_control(dev, kUSB_DeviceControlSetDefaultStatus, NULL);
+		cfg = udc_get_ep_cfg(dev, USB_CONTROL_EP_OUT);
+		if (cfg->stat.enabled) {
+			udc_ep_disable_internal(dev, USB_CONTROL_EP_OUT);
+		}
+		cfg = udc_get_ep_cfg(dev, USB_CONTROL_EP_IN);
+		if (cfg->stat.enabled) {
+			udc_ep_disable_internal(dev, USB_CONTROL_EP_IN);
+		}
+		if (udc_ep_enable_internal(dev, USB_CONTROL_EP_OUT,
+					USB_EP_TYPE_CONTROL,
+					USB_MCUX_EP0_SIZE, 0)) {
+			LOG_ERR("Failed to enable control endpoint");
+			return -EIO;
+		}
+
+		if (udc_ep_enable_internal(dev, USB_CONTROL_EP_IN,
+					USB_EP_TYPE_CONTROL,
+					USB_MCUX_EP0_SIZE, 0)) {
+			LOG_ERR("Failed to enable control endpoint");
+			return -EIO;
+		}
+		udc_submit_event(dev, UDC_EVT_RESET, 0);
+		break;
+	case kUSB_DeviceNotifyError:
+		udc_submit_event(dev, UDC_EVT_ERROR, -EIO);
+		break;
+	case kUSB_DeviceNotifySuspend:
+		udc_set_suspended(dev, true);
+		udc_submit_event(dev, UDC_EVT_SUSPEND, 0);
+		break;
+	case kUSB_DeviceNotifyResume:
+		udc_set_suspended(dev, false);
+		udc_submit_event(dev, UDC_EVT_RESUME, 0);
+		break;
+	case kUSB_DeviceNotifyLPMSleep:
+		break;
+	case kUSB_DeviceNotifyDetach:
+		udc_submit_event(dev, UDC_EVT_VBUS_REMOVED, 0);
+		break;
+	case kUSB_DeviceNotifyAttach:
+		udc_submit_event(dev, UDC_EVT_VBUS_READY, 0);
+		break;
+	default:
+		ep  = mcux_msg->code;
+		if (mcux_msg->isSetup) {
+			struct usb_setup_packet *setup =
+				(struct usb_setup_packet *)mcux_msg->buffer;
+
+			err = udc_mcux_handler_setup(dev, setup);
+		} else if (USB_EP_DIR_IS_IN(ep)) {
+			err = udc_mcux_handler_in(dev, ep, mcux_msg->buffer, mcux_msg->length);
+		} else {
+			err = udc_mcux_handler_out(dev, ep, mcux_msg->buffer, mcux_msg->length);
+		}
+
+		break;
+	}
+
+	if (unlikely(err)) {
+		udc_submit_event(dev, UDC_EVT_ERROR, err);
+		mcux_status = kStatus_USB_Error;
+	}
+	return mcux_status;
+}
+
+static void udc_mcux_isr(const struct device *dev)
+{
+	struct udc_mcux_data *priv = udc_get_private(dev);
+
+	USB_DeviceEhciIsrFunction((void *)(&priv->mcux_device));
+}
+
+/* Return actual USB device speed */
+static enum udc_bus_speed udc_mcux_device_speed(const struct device *dev)
+{
+	int err;
+	uint8_t mcux_speed;
+
+	err = udc_mcux_control(dev, kUSB_DeviceControlGetSpeed, &mcux_speed);
+	if (err) {
+		/* In current all NXP USB device drivers,
+		 * it will not reutrn err if parameter is right
```

Comment:
return?

Should the comment be:
```
/*
 * In the current version of all NXP USB device drivers,
 * no error is returned if the parameter is correct.
 */
```
?

Similar in (udc_mcux_ip3511.c)

Reply:
Update the comments.